providers/sparkpost/src/lib/sparkpost.provider.ts

```diff
@@ -0,0 +1,64 @@
+import {
+  ChannelTypeEnum,
+  ISendMessageSuccessResponse,
+  IEmailOptions,
+  IEmailProvider,
+} from '@novu/stateless';
+import { randomUUID } from 'crypto';
+import SparkPost from 'sparkpost';
+
+export class SparkPostEmailProvider implements IEmailProvider {
+  readonly id = 'sparkpost';
+  readonly channelType = ChannelTypeEnum.EMAIL as ChannelTypeEnum.EMAIL;
+  private readonly client: SparkPost;
+
+  constructor(apiKey: string, eu: boolean) {
+    this.client = new SparkPost(apiKey, {
+      endpoint: eu ? 'https://api.eu.sparkpost.com:443' : undefined,
```

Comment:
Why the other brace of the condition is undefined? Shouldn't be: `https://api.sparkpost.com/api/v1`?

Reply:
Because undefined will default to that string you mentioned automattically as the endpoint is a optional parameter